### PR TITLE
[1LP][RFR] Browser restarter

### PR DESCRIPTION
**Pull request by psav**
Try to restart the browser if it failed during a test ready for the next one

### CODE REVIEW

File: fixtures/browser.py [hunk 1]
```diff
@@ -30,8 +30,28 @@ def pytest_runtest_setup(item):
...
+       isinstance(call.excinfo.value, BadStatusLine) or \
+       isinstance(call.excinfo.value, error):
+        from utils.browser import manager
+        try:
+            if isinstance(manager.factory, WharfFactory):
+                manager.factory.wharf.checkin()
+                manager.factory.wharf.checkout()
+            manager.start()
+            manager.ensure_open()
+        except Exception:
```
Comment (izapolsk) [inline fixtures/browser.py]: excuse me, what is it for ?

File: fixtures/browser.py [hunk 2]
```diff
@@ -30,8 +30,25 @@ def pytest_runtest_setup(item):
...
+    from httplib import BadStatusLine
+    from socket import error
+    from utils.browser import WharfFactory
+
+    import urllib2
+
     name, location = get_test_idents(node)
     val = safe_string(call.excinfo.value.message).decode('utf-8', 'ignore')
+
+    if isinstance(call.excinfo.value, urllib2.URLError) or \
```
Comment (RonnyPfannschmidt) [inline fixtures/browser.py]: use a tuple there

File: fixtures/browser.py [hunk 3]
```diff
@@ -30,8 +30,25 @@ def pytest_runtest_setup(item):
...
+       isinstance(call.excinfo.value, BadStatusLine) or \
+       isinstance(call.excinfo.value, error):
+        from utils.browser import manager
+        if isinstance(manager.factory, WharfFactory):
```
Comment (RonnyPfannschmidt) [inline fixtures/browser.py]: we should move that to the manager itself as a future todo

note that without the url key we may be shooting ourselves for ssui & co

Comment (RonnyPfannschmidt) [inline fixtures/browser.py]: we may get false positives from using rest apis as well

File: fixtures/browser.py (hunk 2 above)
Comment (psav) [inline fixtures/browser.py]: For the exceptions?

File: fixtures/browser.py (hunk 3 above)
Comment (psav) [inline fixtures/browser.py]: False Positives how?

File: fixtures/browser.py (hunk 2 above)
Comment (jkrocil) [inline fixtures/browser.py]:
```
If classinfo is a tuple of class or type objects (or recursively, other such tuples), return true if object is an instance of any of the classes or types.
```
=>
`if isinstance(call.excinfo.value, (urllib2.URLError, BadStatusLine, error)):`

File: fixtures/browser.py (hunk 3 above)
Comment (RonnyPfannschmidt) [inline fixtures/browser.py]: browser restarts due to a rest call failing thats not inside selenium, but any other api using urllib or sockets and not having propper exception layering

**Pull request merged**